cpp/models/ode_secirts/parameters_io.h

```diff
@@ -0,0 +1,1376 @@
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#ifndef ODESECIRTS_PARAMETERS_IO_H
+#define ODESECIRTS_PARAMETERS_IO_H
+
+#include "memilio/config.h"
+
+#ifdef MEMILIO_HAS_JSONCPP
+
+#include "ode_secirts/model.h"
+#include "ode_secirts/analyze_result.h"
+#include "memilio/math/eigen_util.h"
+#include "memilio/mobility/graph.h"
+#include "memilio/mobility/metapopulation_mobility_instant.h"
+#include "memilio/io/epi_data.h"
+#include "memilio/io/io.h"
+#include "memilio/io/json_serializer.h"
+#include "memilio/io/result_io.h"
+#include "memilio/utils/logging.h"
+#include "memilio/utils/date.h"
+
+namespace mio
+{
+namespace osecirts
+{
+
+namespace details
+{
+/**
+        * @brief Reads subpopulations of infection states from transformed RKI cases file.
+        * @param path Path to transformed RKI cases file.
+        * @param vregion vector of keys of the region of interest     
+        * @param date Date for which the arrays are initialized
+        * @param num_* output vector for number of people in the corresponding compartement
+        * @param t_* average time it takes to get from one compartement to another (vector with one element per age group)
+        * @param mu_* probabilities to get from one compartement to another (vector with one element per age group)
+        * @param scaling_factor_inf Factor for scaling the confirmed cases to account for estimated undetected cases.
+        * @see mio::read_confirmed_cases_data
+        * @{
+        */
+IOResult<void> read_confirmed_cases_data(
+    std::string const& path, std::vector<int> const& vregion, Date date, std::vector<std::vector<double>>& num_Exposed,
+    std::vector<std::vector<double>>& num_InfectedNoSymptoms, std::vector<std::vector<double>>& num_InfectedSymptoms,
+    std::vector<std::vector<double>>& num_InfectedSevere, std::vector<std::vector<double>>& num_icu,
+    std::vector<std::vector<double>>& num_death, std::vector<std::vector<double>>& vnum_timm_i,
+    const std::vector<std::vector<int>>& t_Exposed, const std::vector<std::vector<int>>& t_InfectedNoSymptoms,
+    const std::vector<std::vector<int>>& t_InfectedSymptoms, const std::vector<std::vector<int>>& t_InfectedSevere,
+    const std::vector<std::vector<int>>& t_InfectedCritical, const std::vector<std::vector<int>>& vt_imm_interval_i,
+    const std::vector<std::vector<double>>& mu_C_R, const std::vector<std::vector<double>>& mu_I_H,
+    const std::vector<std::vector<double>>& mu_H_U, const std::vector<double>& scaling_factor_inf);
```

Comment:
This interface looks honestly terrifying. Some of the parameters look like they are from the model, can you maybe pass osecirts::parameters instead?

Also, this is quite a big function, and you don't explain what it does, or why. Especially the overload of this is a misnomer, it does not read, instead it computes - something?
Similar for e.g. read_divi_data and read_population_data.

Reply:
good point. This was part of the rework of the file. However, we have nearly the same functions multiple times for the bigger ODE models (secir, secirvvs, secirts). I would leave it in this state since this PR has already enough changes. But after this is merged, i will adress this in issue #1108

Reply:
If you add bullet points to that issue for
- changing the interfaces of these read_sth functions,
- adding some more (code) documentation and
- maybe making them more modular,
then I am fine with that.